### PR TITLE
Add support for Xcode 14 & iOS 16

### PR DESCRIPTION
**Pull request by scannillo**
### Summary of changes

- Add support for Xcode 14 & iOS 16
- Tested all flows on device running iOS 16
- Removed deprecation warnings in Xcode 14

Note: These updates do not drop bitcode support - once we bump the minimum Xcode version to Xcode 14 for merchants, we will drop bitcode support.

### Checklist

- [x] Added a changelog entry

- @scannillo @mattwylder @jaxdesmarais

### CODE REVIEW

File: Sources/BraintreeDataCollector/BTDataCollector.m [hunk 1]
```diff
@@ -56,16 +61,19 @@ - (void)setUpKountWithDebugOn:(BOOL)debugLogging {
     self.kount = [KDataCollector sharedCollector];
     self.kount.debug = debugLogging;
 
-    CLAuthorizationStatus locationStatus = kCLAuthorizationStatusNotDetermined;
+    if ((locationStatus != kCLAuthorizationStatusAuthorizedWhenInUse && locationStatus != kCLAuthorizationStatusAuthorizedAlways) || ![CLLocationManager locationServicesEnabled]) {
```
Comment (scannillo) [inline Sources/BraintreeDataCollector/BTDataCollector.m:64]: 🤔 How do we know that `locationManagerDidChangeAuthorization` will be called to set the `locationStatus` before this method (`setUpKountWithDebugOn`) is called?

Comment (jaxdesmarais) [reply Sources/BraintreeDataCollector/BTDataCollector.m:64]: From the docs for [`locationManagerDidChangeAuthorization`](https://developer.apple.com/documentation/corelocation/cllocationmanagerdelegate/3563956-locationmanagerdidchangeauthoriz) it states:
```
Core Location always calls locationManagerDidChangeAuthorization(_:) when the user’s action results in an authorization status change, and when your app creates an instance of [CLLocationManager](https://developer.apple.com/documentation/corelocation/cllocationmanager), whether your app runs in the foreground or in the background.
```
That said, doing some testing with adding breakpoints this seems somewhat inconsistent on initial load, so we may need to revisit. 🤔

File: CHANGELOG.md [hunk 2]
```diff
@@ -3,6 +3,8 @@
 ## unreleased
 * BraintreeSEPADirectDebit
   * Update nonce to pull in ibanLastFour and customerID as expected
+* Remove unused `presentationContextProvider` (fixes #854)
```
Comment (sshropshire) [inline CHANGELOG.md:6]:
```suggestion
  * Remove unused `presentationContextProvider` (fixes #854)
```

**Review by sshropshire — APPROVED** [review]
🌊

**Review by andrespelaezp — APPROVED** [review]

**Comment by scannillo** [comment]
LGTM!

**Pull request merged**